src/usecases/shlinks/validate-shlink.test.ts

```diff
@@ -0,0 +1,93 @@
+import { SHLinkModel } from "@/domain/models/shlink";
+import { SHLinkValidationError, validateSHLinkUseCase } from "./validate-shlink";
+
+describe('validateSHLinkUseCase', () => {
+  it('should return true for a valid active SHLink with matching passcode', () => {
+    const validSHLink = new SHLinkModel(
+      'unique-user-id',
+      5,
+      true,
+      'management-token',
+      'valid-passcode',
+      new Date(Date.now() + 10000), // future date
+      'link-id'
+    );
+    const result = validateSHLinkUseCase({ shlink: validSHLink, passcode: 'valid-passcode' });
+    expect(result).toBe(true);
+  });
+
+  it('should throw SHLinkValidationError if the SHLink is inactive', () => {
+    const invalidSHLink = new SHLinkModel(
+      'unique-user-id',
+      5,
+      false,
+      'management-token',
+      'valid-passcode',
+      new Date(Date.now() + 10000), // future date
+      'link-id'
+    );
+    expect(() => validateSHLinkUseCase({ shlink: invalidSHLink, passcode: 'valid-passcode' }))
+      .toThrow(SHLinkValidationError);
+  });
+
+  it('should throw SHLinkValidationError if the SHLink is expired', () => {
+    const expiredSHLink = new SHLinkModel(
+      'unique-user-id',
+      5,
+      true,
+      'management-token',
+      'valid-passcode',
+      new Date(Date.now() - 10000), // past date
+      'link-id'
+    );
+    expect(() => validateSHLinkUseCase({ shlink: expiredSHLink, passcode: 'valid-passcode' }))
+      .toThrow(SHLinkValidationError);
```

Comment:
Question/Suggestion. Im wondering if it makes sense to cater for a few different statuses returned within the validation error. The 403 is applicable for most of the validation failures, but wondering around the case where the link is expired, or disabled, that the code should maybe be something like `410 Gone` as this link still exists, but will never be accessible again for retrieval

Reply:
Added 410 for situations where the share link is disabled or expired.